lib/improver/psychrometric_calculations/psychrometric_calculations.py

```diff
@@ -0,0 +1,434 @@
+        """Function to wrap functionality for throwing out temperatures
+        too low or high for a method to use safely.
+
+        Args:
+            cube (iris.cube.Cube):
+                A cube of temperature.
+
+            low (int or float):
+                Lowest allowable temperature for check
+
+            high (int or float):
+                Highest allowable temperature for check
+
+        Raises:
+            UserWarning : If any of the values in cube.data are outside the
+                          bounds set by the low and high variables.
+        """
+        if cube.data.max() > high or cube.data.min() < low:
+            emsg = ("Wet bulb temperatures are being calculated for conditions"
+                    " beyond the valid range of the saturated vapour pressure"
+                    " lookup table (< {}K or > {}K). Input cube has\n"
+                    "Lowest temperature = {}\nHighest temperature = {}")
+            warnings.warn(emsg.format(low, high, cube.data.min(),
+                                      cube.data.max()))
+
+    def _lookup_svp(self, temperature):
+        """
+        Looks up a value for the saturation vapour pressure of water vapour
+        using the temperature and a table of values. These tabulated values
+        have been calculated using the utilties.ancillary_creation
+        SaturatedVapourPressureTable plugin that uses the Goff-Gratch method.
+
+        Args:
+            temperature (iris.cube.Cube):
+                A cube of air temperatures (K).
+        Returns:
+            svp (iris.cube.Cube):
+                A cube of saturated vapour pressures (Pa).
+        """
+        T_min = svp_table.T_MIN
+        T_max = svp_table.T_MAX
+        delta_T = svp_table.T_INCREMENT
+        self._check_range(temperature, T_min, T_max)
+        temperatures = temperature.data
+        T_clipped = np.clip(temperatures, T_min, T_max)
+
+        # Note the indexing below differs by -1 compared with the UM due to
+        # Python vs. Fortran indexing.
+        table_position = (T_clipped - T_min + delta_T)/delta_T - 1.
+        table_index = table_position.astype(int)
+        interpolation_factor = table_position - table_index
+        svps = ((1.0 - interpolation_factor) * svp_table.DATA[table_index] +
+                interpolation_factor * svp_table.DATA[table_index + 1])
+
+        svp = temperature.copy(data=svps)
+        svp.units = Unit('Pa')
+        svp.rename("saturated_vapour_pressure")
+        return svp
+
+    @staticmethod
+    def _pressure_correct_svp(svp, temperature, pressure):
+        """
+        Convert saturated vapour pressure in a pure water vapour system into
+        the saturated vapour pressure in air.
+
+        Args:
+            svp (iris.cube.Cube):
+                A cube of saturated vapour pressures (Pa).
+            temperature (iris.cube.Cube):
+                A cube of air temperatures (K, converted to Celsius).
```

Comment:
It's a private function. It won't be called by any unknown entity and I perform the unit conversion in calculate_wet_bulb_temperature. The Celsius conversion is there because I know it comes in in K.

Reply:
Ok.